app/app/me/components/ProfileEditor.tsx

```diff
@@ -1,142 +1,50 @@
-  PlusIcon,
-  TrashIcon,
 } from "@heroicons/react/24/outline";
-
-const NEW_JOB_EXPERIENCE = {
-  employer: "",
-  position: "",
-  date_from: "",
-  date_to: "",
-};
-
-const SOCIAL_NETWORKS_TYPES = [
-  "Slack",
-  "LinkedIn",
-  "GitHub",
-  "Phone",
-  "Instagram",
-  "Telegram",
-  "Discord",
-  "Other",
-].map((type) => ({ key: type, value: type }));
-
-type JobHistoryType = {
-  employer?: string | null;
-  position?: string | null;
-  date_from?: Date | string | null;
-  date_to?: Date | string | null;
-};
-
-type SocialNetworkType = {
-  type?: string | null;
-  link?: string | null;
-};
-
-type ProfileFormData = {
-  profile_picture?: string | null;
-  first_name: string;
-  last_name: string;
-  nationality?: string | null;
-  university?: string | null;
-  degree_level?: string | null;
-  degree_name?: string | null;
-  degree_semester?: number | null;
-  currentJob?: string | null;
-  description?: string | null;
-  job_history: JobHistoryType[];
-  social_networks: SocialNetworkType[];
-};
-
-const validationSchema = Yup.object().shape({
-  profile_picture: Yup.mixed()
-    .test("fileSize", "File is too big! Max size is 200 KB.", (value) => {
-      if (value instanceof FileList) {
-        return value[0]?.size <= 0.2 * 1048576;
-      }
-      return true; // not a file input
-    })
-    .nullable(),
-  first_name: Yup.string().required("First name is required."),
-  last_name: Yup.string().required("Last name is required."),
-  nationality: Yup.string().nullable(),
-  university: Yup.string().nullable(),
-  degree_level: Yup.string().nullable(),
-  degree_name: Yup.string().nullable(),
-  degree_semester: Yup.number()
-    .min(1, "Semester must be at least 1.")
-    .integer("Semester must be an integer.")
-    .nullable(),
-  currentJob: Yup.string().nullable(),
-  description: Yup.string().nullable(),
-  job_history: Yup.array().of(
-    Yup.object().shape({
-      employer: Yup.string().required("Employer is required."),
-      position: Yup.string().required("Position is required."),
-      date_from: Yup.date().required("Start date is required."),
-      date_to: Yup.date()
-        .min(Yup.ref("date_from"), "End date must be after the start date.")
-        .required("End date is required."),
-    }),
-  ),
-  social_networks: Yup.array().of(
-    Yup.object().shape({
-      type: Yup.string().required("Type is required."),
-      link: Yup.string()
-        .url("Invalid URL format")
-        .required("Link is required."),
+import {UploadIcon, DownloadIcon} from "@radix-ui/react-icons";
+import { z } from 'zod';
+import { MeModel } from "@models/me";
+
+const profileSchema = z.object({
+  /*
+  profile_picture: z.union([z.string().nullable(), z.instanceof(FileList).optional()])
+    .refine((file) => !file || (file instanceof FileList && file[0].size <= 0.2 * 1048576), {
+      message: "File is too big! Max size is 200 KB.",
     }),
-  ),
+  */
+  first_name: z.string().nonempty("First name is required."),
```

Comment:
Whats the difference between `.nonempty` and `.required?`

Reply:
Thanks a lot for switching to zod with react-forms from Formik! Very clean

Reply:
nonempty prevents empty inputs, e.g. an empty string, which "required" doesn't necessarily do